src/pure.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { ReactElement, ReactNode, Suspense } from 'react'
+import { act, create, ReactTestRenderer } from 'react-test-renderer'
+import asyncUtils from './asyncUtils'
+import { cleanup, addCleanup, removeCleanup } from './cleanup'
+
+// TODO: Add better type, currently file is utilizing minimum types to work
+// TODO: Attempt to refactor code to remove ESLint disables if possible
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Props<T = any, R = any> = {
+  callback: (props: T) => R
+  hookProps: unknown
+  onError: CallableFunction
+  children: CallableFunction
+}
+function TestHook({ callback, hookProps, onError, children }: Props) {
```

Comment:
I'm not sure if this needs to be generic but we are not calling passing anything to it so it will always be any

Reply:
@merodiro It should be inferred from the callback wich is passed on line 69:
![image](https://user-images.githubusercontent.com/40751395/101999395-bf2c2d00-3d30-11eb-9c17-bcf20714ffdd.png)

Reply:
`<TestHook>` isn't generic and isn't passing anything to `Props` so it will always be any

Reply:
@merodiro Oh 🤦‍♂️ I completely missed that

Reply:
I think `TestHook` should be made generic.